File: src/main/java/cookiedragon/luchadora/module/impl/player/NoEntityTraceModule.java
package cookiedragon.luchadora.module.impl.player;

import cookiedragon.luchadora.event.api.Subscriber;
import cookiedragon.luchadora.event.entity.GetEntitiesMouseOverEvent;
import cookiedragon.luchadora.module.AbstractModule;
import cookiedragon.luchadora.module.Category;

/**
 * @author cookiedragon234 02/Jan/2020
 */
@AbstractModule.Declaration(name = "No Entity Trace", description = "You can hit blocks through entities", category = Category.PLAYER)
public class NoEntityTraceModule extends AbstractModule
{
	@Subscriber
	private void onEntityTrace(GetEntitiesMouseOverEvent event)
	{
		event.size = 0;
	}
}

File: src/main/kotlin/cookiedragon/luchadora/module/impl/player/NoEntityTraceModule.kt
package cookiedragon.luchadora.module.impl.player

import cookiedragon.eventsystem.Subscriber
import cookiedragon.luchadora.event.entity.GetEntitiesMouseOverEvent
import cookiedragon.luchadora.module.AbstractModule
import cookiedragon.luchadora.module.Category

/**
 * @author cookiedragon234 12/Jan/2020
 */
class NoEntityTraceModule : AbstractModule("No Entity Trace", "You can hit blocks through entities", Category.PLAYER) {
	@Subscriber
	private fun onEntityTrace(event: GetEntitiesMouseOverEvent) {
		event.size = 0
	}
}
